hazelcast/src/test/java/com/hazelcast/map/ComputeConditionallyTests.java

```diff
@@ -0,0 +1,159 @@
+import java.util.function.BiFunction;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ComputeConditionallyTests extends HazelcastTestSupport {
+
+    private final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+
+    private HazelcastInstance server;
+    private HazelcastInstance client;
+
+    @Before
+    public void setup() {
+        ClassLoader customClassLoader = new ClassLoader(Thread.currentThread().getContextClassLoader().getParent()) {
+        };
+        Config config = getConfig();
+        config.setClassLoader(customClassLoader);
+        config.getUserCodeDeploymentConfig().setEnabled(true);
+        server = hazelcastFactory.newHazelcastInstance(config);
+
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getUserCodeDeploymentConfig().setEnabled(true);
+        clientConfig.getUserCodeDeploymentConfig().addClass(StaticSerializableBiFunction.class);
+        clientConfig.getUserCodeDeploymentConfig().addClass(StaticNonSerializableBiFunction.class);
+        client = hazelcastFactory.newHazelcastClient(clientConfig);
+    }
+
+    @After
+    public void tearDown() {
+        hazelcastFactory.terminateAll();
+    }
+
+    @Test
+    public void testComputeIfPresentWithLambdaReMappingFunction() {
+        final String outer_state = "outer_state";
+        BiFunction biFunction = (key, oldValue) -> "new_value_from_lambda_and_" + outer_state;
+        testComputeIfPresentForFunction(biFunction, "new_value_from_lambda_and_outer_state");
+    }
+
+
+    @Test
+    public void testComputeIfPresentWithAnonymousReMappingFunction() {
+        BiFunction<String, String, String> biFunction = NonStaticFunctionFactory
+                .getAnonymousNonSerializableBiFunction("new_value");
+        testComputeIfPresentForFunction(biFunction, "new_value");
+    }
+
+    @Test
+    public void testComputeIfPresentWithStaticSerializableRemappingFunction() {
+        StaticSerializableBiFunction biFunction = new StaticSerializableBiFunction("new_value");
+        testComputeIfPresentForFunction(biFunction, "new_value");
+    }
+
+    @Test
+    public void testComputeIfPresentWithStaticNonSerializableRemappingFunction() {
+        StaticNonSerializableBiFunction biFunction = new StaticNonSerializableBiFunction("new_value");
+        testComputeIfPresentForFunction(biFunction, "new_value");
+    }
+
+    @Test(expected = ClassNotFoundException.class)
+    public void testComputeIfPresentWithRemappingFunctionPresentOnClientJVMOnly() throws Throwable {
+        ClientOnlyStaticSerializableBiFunction biFunction = new ClientOnlyStaticSerializableBiFunction("new_value");
+        try {
+            testComputeIfPresentForFunction(biFunction, "new_value");
+        } catch (HazelcastSerializationException e) {
```

Comment:
You can use `com.hazelcast.internal.util.RootCauseMatcher` instead. See [here](https://github.com/hazelcast/hazelcast/blob/97ff57ba92dc1fbfa77720885e7db40757999080/hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/ClientInvocationFutureTest.java#L112-L119) for an example.